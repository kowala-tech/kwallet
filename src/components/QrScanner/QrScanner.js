import React from "react";
import PropTypes from "prop-types";
import { compose } from "recompose";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
import Drawer from "@material-ui/core/Drawer";
import Button from "@material-ui/core/Button";
// Component Related Imports
import styles from "./styles";
import QrReader from "react-qr-reader";

class QrScanner extends React.Component {
	constructor(props){
		super(props);
		this.state = {
			delay: 300,
			open: false
		};
	}

	handleError(err){
		console.error(err);
	}

	toggle(){
		this.setState({ open: !this.state.open });
	}


	render(){

		const {
			handleScan
		} = this.props;

		const scanAndClose = (result) => {
			if (result && result.length > 10) {
				handleScan(result) && this.toggle;
			}
		};

		return(
			<div>
				<Button
					variant="raised"
					fullWidth
					color="primary"
					onClick={this.toggle.bind(this)}
				>
					Scan Wallet Address QR Code
				</Button>
				<Drawer
					anchor="bottom"
					open={this.state.open}
					onClose={this.toggle.bind(this)}
				>
					<div
						role="button"
						onClick={this.toggle.bind(this)}
						onKeyDown={this.toggle.bind(this)}
					>
						<QrReader
							showViewFinder={false}
							delay={this.state.delay}
							onError={this.handleError}
							onScan={(result) => scanAndClose(result)}
							style={{ width: "100%" }}
						/>
					</div>
				</Drawer>
			</div>
		);
	}
}

QrScanner.propTypes = {
	classes: PropTypes.object.isRequired,
	handleScan: PropTypes.func.isRequired
};

export default compose(
	withStyles(styles),
)(QrScanner);
