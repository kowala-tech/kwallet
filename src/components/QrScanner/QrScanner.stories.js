import React from "react";
import { storiesOf } from "@storybook/react";
// import { action } from "@storybook/addon-actions";
// Story related component imports
import QrScanner from "./QrScanner";

storiesOf("QR Scanner", module)
	.add("default", () =>
		(<QrScanner/>)
	);
