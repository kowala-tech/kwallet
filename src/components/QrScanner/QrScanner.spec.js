import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import CameraIcon from "@material-ui/icons/Camera";
import QrScanner from "./QrScanner";

describe("QrScanner", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<QrScanner
					handleScan={jest.fn()}
					buttonIcon={<CameraIcon />}
				/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
