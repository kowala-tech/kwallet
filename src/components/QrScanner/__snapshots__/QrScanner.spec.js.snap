// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QrScanner it should match the snapshot 1`] = `
<QrScanner
  buttonIcon={<pure(Camera) />}
  classes={Object {}}
  handleScan={[MockFunction]}
/>
`;
